pkg/image/prune/imagepruner.go

```diff
@@ -488,10 +557,33 @@ func (p *imagePruner) Run(pruneImage ImagePruneFunc, pruneStream ImageStreamPrun
 	allNodes := p.g.NodeList()
 
 	imageNodes := imageNodeSubgraph(allNodes)
+	if len(imageNodes) == 0 {
+		return
+	}
+
+	// we only support a single internal registry, and all images should have the same registry
+	// so we just take the 1st one and use it
+	ref, err := imageapi.ParseDockerImageReference(imageNodes[0].Image.DockerImageReference)
+	if err != nil {
+		glog.Errorf("error determining registry URL from %q: %v", imageNodes[0].Image.DockerImageReference, err)
+		return
+	}
+
+	registryURL := ref.Registry
+	if len(registryURL) == 0 {
+		glog.Errorf("unable to prune - registry URL is missing")
+		return
+	}
+
+	if err := p.registryPinger.ping(registryURL); err != nil {
+		glog.Errorf("error communicating with registry: %v", err)
+		return
+	}
+
 	pruneImages(p.g, imageNodes, pruneImage, pruneStream, pruneManifest)
 
 	layerNodes := layerNodeSubgraph(allNodes)
-	pruneLayers(p.g, layerNodes, pruneLayer, pruneBlob)
+	pruneLayers(p.g, registryURL, layerNodes, pruneLayer, pruneBlob)
```

Comment:
i think this highlights why we have to have a registry prune function - this is very unsafe, mostly because any code fault or interruption leaves you with dangling layers.  We should be purging before deletion, not after, while tombstoning.  I'd like to have the larger discussion before merging this, because this seems to exacerbate prune problems.